src/lib/kadm5/kadm_rpc.h

```diff
@@ -108,6 +108,15 @@ struct setkey3_arg {
 };
 typedef struct setkey3_arg setkey3_arg;
 
+struct setkey4_arg {
+	krb5_ui_4 api_version;
+	krb5_principal princ;
+	krb5_boolean keepold;
+	kadm5_key_data *key_data;
+	int n_key_data;
+};
```

Comment:
So, Nico is advocating the addition of an extra flag argument, in anticipation of future support for a kvno tl-data item in principal entries.  The purpose of the anticipated kvno tl-data item is to make the KDC continue serving existing keys for a window of time until we are sure that new keys have been stored in all of the relevant keytabs.  I tentatively agree with Simo that adding this argument is not warranted.  Further thoughts:

* If we decide to add such an argument for setkey, we probably need it for chrand as well, and possibly for chpass.

* I can think of a plausible design which does not need this argument.  To wit: if a kvno tl-data value is present, it remains unchanged by setkey/chpass/chrand when keepold is set, and you just have to update it explicitly afterwards.